cassandra/src/main/scala/filodb.cassandra/columnstore/IngestionTimeIndexTable.scala

```diff
@@ -63,36 +82,30 @@ sealed class IngestionTimeIndexTable(val dataset: DatasetRef, val connector: Fil
   def scanRowsByIngestionTimeNoAsync(tokens: Seq[(String, String)],
                                      ingestionTimeStart: Long,
                                      ingestionTimeEnd: Long): Iterator[Row] = {
-    def cql(start: String, end: String): String =
-      s"SELECT partition, ingestion_time, start_time, info FROM $tableString " +
-      s"WHERE TOKEN(partition) >= $start AND TOKEN(partition) < $end " +
-      s"AND ingestion_time >= $ingestionTimeStart AND ingestion_time <= $ingestionTimeEnd " +
-      s"ALLOW FILTERING"
 
     tokens.iterator.flatMap { case (start, end) =>
-      session.execute(cql(start, end)).iterator.asScala
+      val stmt = scanCql1.bind(start.toLong: java.lang.Long,
+                               end.toLong: java.lang.Long,
+                               ingestionTimeStart: java.lang.Long,
+                               ingestionTimeEnd: java.lang.Long)
```

Comment:
@velvia  one confirmation I wanted from you is that for token scans, we were using String parameter from before for the Cassandra start/end tokens. I never understood why. Conversion of token to Long should not be an issue right ?

Reply:
In other words, tokens should be within limits of Long ?

Reply:
Confirmed that this works fine or Murmur3Partitioner tokens - which is why existing approach of converting to strings has been working.

Reply:
Was this validated through tests?   Long _should_ be fine, though tokens can wrap also.  As long as we aren't doing manipulation of the tokens it's probably fine, but Datastax driver actually has Token objects.  I would think they would let you bind to the Token objects directly, that would be preferable to using longs

Reply:
I did try Token objects, but they are not Serializable. It breaks our spark jobs, and I dont want to open a can of worms there.